### PR TITLE
Issue #3153 - Minimap corrections

### LINKED ISSUE #3153: Issue: Mini Map Graphics Errors and Repaint Glitch
### Environment
Starting MegaMek v0.49.3 ...
Today is 2021-09-13
Java vendor Eclipse Foundation
Java version 11.0.12
Platform Windows 10 10.0 (amd64)
Total memory available to MegaMek: 1,048,576 kB

### Description
Setting up a simple test scenario with an Archer and a VTOL on a single or double map, I bring up the mini map and start clicking the zoom + and zoom minus and center arrow button (green bar below the minimap).  Sometimes when clicking the green bar toward the left or right edge, the mini-map is replaced with some graphics from elsewhere on the screen.  You can click and scroll on the mini map, but instead of repainting the mini map, other background graphics are painted.

MiniMap
MiniMap2
MiniMap3

### Files
[megameklog.zip](https://github.com/MegaMek/mekhq/files/7158181/megameklog.zip)

### PR DESCRIPTION
**Pull request by SJuliez**
Some drawing errors as mentioned in the issue. Also corrects some other errors (the flip mousewheel behavior was all wrong) and a few minor changes as suggested by intellij warnings.

Resolves #3153

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/MiniMap.java [hunk 1]
```diff
@@ -404,7 +394,7 @@ void initializeMap() {
         leftMargin = margin;
         int requiredWidth = (board.getWidth() * (HEX_SIDE[zoom] + HEX_SIDE_BY_SIN30[zoom]))
                 + HEX_SIDE_BY_SIN30[zoom] + (2 * margin);
-        int requiredHeight = (((2 * board.getHeight()) + 1)
+        int requiredHeight = minimized ? 14 : (((2 * board.getHeight()) + 1)
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/MiniMap.java:397]: 14?

File: megamek/src/megamek/client/ui/swing/MiniMap.java [hunk 2]
```diff
@@ -1267,57 +1255,57 @@ private void zoomIn() {
...
+                || (y <= getSize().height - buttonHeight)) {
+            return;
+        }
+        if (minimized) {
+            setSize(getSize().width, heightBuffer);
+            mapImage = ImageUtil.createAcceleratedImage(getSize().width, heightBuffer);
+            minimized = false;
+            initializeMap();
+        } else {
+            if (x < 14) {
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/MiniMap.java:1291]: 14 AND 28 AND 2!

All joking aside, can you toss in a little comment explaining what these 14s and 28s are?

**Review by NickAragua — CHANGES_REQUESTED** [review]
Changes look good as far as I can tell, although I'd appreciate a couple of explanatory comments about some particularly mysterious magic numbers.

File: megamek/src/megamek/client/ui/swing/MiniMap.java [hunk 3]
```diff
@@ -1435,37 +1423,32 @@ private void update() {
...
-            processMouseClick(e.getX(), e.getY(), e);
+        public void mouseDragged(MouseEvent me) {
+            Point mapPoint = SwingUtilities.convertPoint(dialog, me.getX(), me.getY(), MiniMap.this);
+            processMouseDrag(mapPoint.x, mapPoint.y, me.getModifiersEx());
         }
     };
 
     MouseWheelListener mouseWheelListener = (we) -> {
-        if (we.getWheelRotation() < 0) {
+        if (we.getWheelRotation() > 0 ^ GUIP.getMouseWheelZoomFlip()) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MiniMap.java:1445]: Zero raised to a power?

Comment (NickAragua) [reply megamek/src/megamek/client/ui/swing/MiniMap.java:1445]: Think that's an XOR

Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/MiniMap.java:1445]: You're correct, thanks

Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/MiniMap.java:1445]: Raising a number to a boolean, that would be interesting. Two to the truth?

**Comment by SJuliez** [comment]
OK, one more try: Replaced many numbers with BUTTON_HEIGHT. There's more +3 and -6 stuff, but thats pixel drawing and imo not worth replacing. Also some more changes to the mouse listeners to limit the reaction to inside the minimap (the events fire also slightly outside the dialog) and to improve the Ctrl-Click reaction with dragging.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**